### PR TITLE
Issues/7 static code analysis workflows

### LINKED ISSUE #7: Static Code analysis and Github Workflows
* Add static code analysis maven plugin
* Update Github workflows to only run once in pull requests

### PR DESCRIPTION
**Pull request by wetret**
resolves #7

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -39,13 +39,13 @@
...
 			<artifactId>dsf-bpe-process-api-v1</artifactId>
-			<version>1.3.1</version>
+			<version>1.5.0</version>
 			<scope>provided</scope>
 		</dependency>
 		<dependency>
 			<groupId>de.hs-heilbronn.mi</groupId>
 			<artifactId>crypto-utils</artifactId>
-			<version>3.8.0</version>
+			<version>4.0.0</version>
```
Comment (hhund) [inline pom.xml:48]: DSF process api v1 useses crypto utils 3.8.0. do not upgrade to 4.0.0 unless you include it in a fat process plugin jar.

```suggestion
			<version>3.8.0</version>
```

File: .github/workflows/maven-build.yml [hunk 2]
```diff
@@ -1,23 +1,25 @@
...
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: mvn -B verify -fae --file pom.xml
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          cache: 'maven'
+      - uses: s4u/maven-settings-action@v2.8.0
```
Comment (hhund) [inline .github/workflows/maven-build.yml:18]: Consider referencing a specific commit hash for third-party or even all GitHub actions, https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

> **Pin actions to a full length commit SHA**
>
> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload. When selecting a SHA, you should verify it is from the action's repository and not a repository fork.

```suggestion
      - uses: s4u/maven-settings-action@60912582505985be4cc55d2b890eb32767f8de5f #v2.8.0
```
...there is a v3.0.0 release

File: .github/workflows/maven-publish.yml [hunk 3]
```diff
@@ -1,32 +1,31 @@
...
+      - uses: actions/checkout@v4
       - name: Set up JDK 17
-        uses: actions/setup-java@v3
+        uses: actions/setup-java@v4
         with:
-          distribution: 'temurin'
+          distribution: 'zulu'
           java-version: 17
+          cache: 'maven'
+      - uses: s4u/maven-settings-action@v2.8.0
```
Comment (hhund) [inline .github/workflows/maven-publish.yml:24]: see maven-build.yml

File: src/main/java/de/medizininformatik_initiative/processes/common/fhir/client/FhirClientFactory.java [hunk 4]
```diff
@@ -16,10 +16,10 @@
 import org.slf4j.LoggerFactory;
 
 import ca.uhn.fhir.context.FhirContext;
+import de.hsheilbronn.mi.utils.crypto.CertificateHelper;
+import de.hsheilbronn.mi.utils.crypto.io.CertificateReader;
+import de.hsheilbronn.mi.utils.crypto.io.PemIo;
```
Comment (hhund) [inline src/main/java/de/medizininformatik_initiative/processes/common/fhir/client/FhirClientFactory.java:21]: crypto-utils 4.0.0 can only be used if copied into the process plugin, see pom.xml comment

File: pom.xml (hunk 1 above)
Comment (wetret) [reply pom.xml]: reset with bf0f632

File: src/main/java/de/medizininformatik_initiative/processes/common/fhir/client/FhirClientFactory.java (hunk 4 above)
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/processes/common/fhir/client/FhirClientFactory.java]: reset with bf0f632

File: .github/workflows/maven-build.yml (hunk 2 above)
Comment (wetret) [reply .github/workflows/maven-build.yml]: implemented with 647c5e5

File: .github/workflows/maven-publish.yml (hunk 3 above)
Comment (wetret) [reply .github/workflows/maven-publish.yml]: implemented with 647c5e5

**Review by hhund — APPROVED** [review]

**Pull request merged**